code/modules/spells/spell_types/wizard/invoked_single_target/dominate_mind.dm

```diff
@@ -0,0 +1,244 @@
+
+/datum/component/dominated_mind/proc/confirm_submission()
+	var/mob/living/puppet = parent
+	if(!puppet || !master)
+		return
+
+	var/confirmation = alert(puppet, "Are you sure you want to permanently submit your will to [master.real_name]? This is irreversible and will last until death.", "Confirm Submission", "Yes", "No")
+
+	if(QDELETED(src) || QDELETED(puppet) || QDELETED(master))
+		return
+
+	if(confirmation == "Yes")
+		make_permanent()
+
+/datum/component/dominated_mind/proc/on_master_speak(mob/living/speaker, list/speech_args)
+	SIGNAL_HANDLER
+	var/mob/living/puppet = parent
+
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(!message)
+		return
+	message = lowertext(message)
+
+	if(!master || QDELETED(puppet) || puppet.stat != CONSCIOUS)
+		qdel(src)
+		return
+
+	var/is_yell = (say_test(message) >= "2")
+	// Find a command that matches the message and invocation type
+	if(command_cooldown && world.time < command_cooldown)
+		to_chat(master, span_warning("You need more time to focus your will before issuing another command."))
+		return
+	for(var/datum/voice_of_god_command/command as anything in GLOB.voice_of_god_commands)
+		if(findtext(message, command.trigger))
+			if(command.tier > enthrallment_level)
+				to_chat(master, span_warning("Your psychic connection with [puppet.real_name] is not yet strong enough to compel such a command."))
+				return
+
+			if(command.tier == 3)
+				if(!is_yell)
+					to_chat(master, span_warning("You must exert your will more forcefully to make them obey that command!"))
+					return 
+			if(command.tier >= 2 && !is_overt)
+				to_chat(puppet, span_mind_control("You are overwhelmed by a controlling presence throughout your mind!"))
+				// Add the submission verb to the puppet
+				var/mob/living/carbon/human/slave = parent
+				if(slave)
+					slave.verbs += /mob/living/proc/submit_to_master
+
+			// Execute the command on the puppet
+			command.execute(list(puppet), master, 1, message)
+
+			// Set the cooldown for this command type
+			if(command.cooldown > 0)
+				to_chat(master, span_warning("You must recover for [DisplayTimeText(command.cooldown)] seconds before issuing another command."))
+				command_cooldown = world.time + command.cooldown
+			return
+
+/datum/component/dominated_mind/proc/handle_hearing(datum/source, list/hearing_args)
+	SIGNAL_HANDLER
+
+	var/mob/living/puppet = parent
+	var/mob/living/speaker = hearing_args[HEARING_SPEAKER]
+
+	// If the speaker isn't our master, or if the speaker is invalid, we do nothing.
+	if(!speaker || speaker != master)
+		return
+
+	// Don't reformat the puppet's own speech
+	if(puppet == speaker)
+		return
+
+	// Reformat the raw message text.
+	hearing_args[HEARING_RAW_MESSAGE] = span_phobia(hearing_args[HEARING_RAW_MESSAGE])
+
+/// This verb allows a player under the effect of Dominate Mind to permanently submit to their master.
+/mob/living/proc/submit_to_master()
```

Comment:
this verb appears to be available to non-dominated players 

<img width="681" height="1439" alt="Image" src="https://github.com/user-attachments/assets/273e0978-0602-451a-9751-c74b64915bef" />